Shopify/cli
# Adopt the CLI UI kit in the `shopify theme open` command

### WHY are these changes introduced?

Fixes:
- https://github.com/Shopify/internal-cli-foundations/issues/496
- https://github.com/Shopify/internal-cli-foundations/issues/497

### WHAT is this pull request doing?

This PR has two main parts, the REST Admin API client and the `shopify theme open` changes. Let's approach these changes by taking a look at modified files:

- **`.vscode/launch.json`** - it introduces a handy debugger for themes

- **`packages/cli-kit/src/api/admin*`** - here we're introducing the thin client to handle REST API calls, at the [@shopify/cli-kit](https://github.com/Shopify/cli/blob/main/docs/cli/architecture.md#modules-packages) level

- **`packages/theme/src/cli/utilities/themes-api*`** - here we're introducing themes API-specific logic

- **`packages/theme/src/cli/commands/theme/open*`**
  **`packages/theme/src/cli/services/open*`** - the `shopify theme open` command was updated following the [convention](https://github.com/Shopify/cli/blob/main/docs/cli/conventions.md#naming-and-directory-conventions-1) of having a service module with the same name

- **`packages/theme/src/cli/models/theme.ts`** - it's a thin data-structure that represents theme

- **`packages/theme/src/cli/utilities/theme-selector*`**
- **`packages/theme/src/cli/utilities/theme-urls*`** - here we're introducing utility functions to support our `services`

### How to test your changes?

- Run `pnpm run shopify theme open -t <theme ID>`
- Run `pnpm run shopify theme open -t <theme name>`
- Run `pnpm run shopify theme open -l`
- Run `pnpm run shopify theme open -d`
- Run `pnpm run shopify theme open`

**Notice**: The [prototype](https://user-images.githubusercontent.com/1079279/210743403-7a41ab62-56ef-43f9-af8c-14444f052857.png) doesn't have a header, however the CLI UI kit API requires it. I'm checking with DX, so feel free to ignore that detail for now :)

**Before**
![Screenshot 2023-01-05 at 14 33 09](https://user-images.githubusercontent.com/1079279/210792291-3482c9d2-5536-451a-a985-0ed3fe21f3ce.png)


**After**
![Screenshot 2023-01-05 at 14 34 09](https://user-images.githubusercontent.com/1079279/210792306-5d9f979e-03b8-4c1b-8479-6cebd55cf69a.png)


### Post-release steps

None.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
